discv5/discv5-wire.md

```diff
@@ -281,7 +181,33 @@ current record as the only result.
 NODES is the response to a FINDNODE or TOPICQUERY message. Multiple NODES messages may be
 sent as responses to a single query.
 
-### REGTOPIC Request (0x05)
+When handling NODES as a response to FINDNODE, the recipient should verify that the
+received nodes match the requested distances.
+
+### TALKREQ Request (0x05)
+
+    message-data = [request-id, protocol, request]
+    message-type = 0x09
+
+TALKREQ sends an application-level request. The purpose of this message is pre-negotiating
+connections made through another application-specific protocol identified by `protocol`.
+
+If `request-id` is non-zero, the recipient must respond with a TALKRESP message containing
+the response to the request. If the `protocol` is unknown to the recipient, it must
+respond with an empty TALKRESP response.
+
+### TALKRESP Response (0x06)
+
+    message-data = [request-id, response]
```

Comment:
What do you think about adding a `total` field to this message so that it is easier to implement multi-message responses naively.  Without this it appears that value will have to be embedded in the `response` itself which makes it awkward to fit this into the same multi-response framework that the other messages use.

Reply:
Hmm, I didn't consider that anyone could want multiple responses for this request.

Reply:
My experience with UDP is limited so I may be leaning the wrong way here.  My thinking here is base on:

1. The `Nodes` response message which allows the payload to be distributed across multiple responses.
2. Planning to implement distributed storage of chain data via `TALKREQ/TALKRESP` in which the payloads will be larger than can fit into a single datagram requiring them to be split across multiple messages.

That said, it may be inappropriate to implement that at this level so maybe hold off until I actually get around to using them and i can re-raise this then if it still makes sense.